### PR TITLE
Improve processors context

### PR DESCRIPTION
**Pull request by blocknotes**
Minor refactor to processors to remove the unused keys from processors context and replace the failure checks with exceptions.

### CODE REVIEW

File: lib/solidus_importer/base_importer.rb [hunk 1]
```diff
@@ -2,8 +2,17 @@
...
   class BaseImporter
+    attr_reader :options, :logger
+
     def initialize(options)
       @options = options
+      @logger ||= begin
+        if @options[:logger] && !@options[:logger].is_a?(Class)
+          @options[:logger]
+        else
+          (@options[:logger] || ::SolidusImporter::BaseLogger).new
```
Comment (elia) [inline lib/solidus_importer/base_importer.rb]: Any reason for not passing `Logger.new(STDOUT)` directly?

File: lib/solidus_importer/process_row.rb [hunk 2]
```diff
@@ -12,12 +12,17 @@ def initialize(importer, row)
...
+      @importer.processors.each do |processor|
+        begin
+          processor.call(context)
+        rescue => e # rubocop:disable Style/RescueStandardError
+          @importer.process_row_failure(processor, context: context, error: e)
+          context.merge!(success: false, messages: e.message) # rubocop:disable Performance/RedundantMerge
+          break
+        end
       end
+      @importer.logger.info ::SolidusImporter::Log.new(source: @row, context: context)
```
Comment (elia) [inline lib/solidus_importer/process_row.rb]: Is it possible to pass the source to the logger when initializing it and only send strings as logs?

File: lib/solidus_importer/base_importer.rb (hunk 1 above)
Comment (blocknotes) [reply lib/solidus_importer/base_importer.rb]: Good point 👍 

As discussed, I'm going to move the logger replacement in another PR.

Comment (blocknotes) [reply lib/solidus_importer/base_importer.rb]: Moved [here](https://github.com/nebulab/solidus_importer/pull/19)

**Pull request merged**